packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -396,6 +400,18 @@ export class DataStores implements IDisposable {
                 return context.attachState === AttachState.Attached;
             }).map(async ([contextId, context]) => {
                 const contextSummary = await context.summarize(fullTree, trackState);
+                const summary = contextSummary.summary;
+                if (summary.type === SummaryType.Tree && summary.unreferenced === true) {
+                    summarizedUnreferencedDataStoreCount++;
+                }
+                if (summary.type === SummaryType.Handle) {
+                    handleCount++;
+                }
+                dataStoreCount++;
+
+                contextSummary.stats.dataStoreCount = dataStoreCount;
```

Comment:
This seems incorrect. You should add these stats after the Promise.all is completed to the summary tree that is returned by summarize.